### PR TITLE
Split into 2 apps

### PR DESCRIPTION
**Pull request by cb372**
`recommend-backend` consumes events from Ophan and calculates the recommendations for each variant. It uploads its results to S3 periodically.

`recommend-frontend` handles requests from the website frontend. It combines recommendations (which it loads by polling the files on S3) with CAPI content.

There is also a trivial `common` project containing shared model classes.

This PR also updates `cloudformation.json` to define both the backend and frontend stacks.

### CODE REVIEW

File: frontend/app/com/gu/recommend/capi/Capi.scala [hunk 1]
```diff
@@ -65,6 +65,7 @@ class Capi(capiKey: String, ws: WSAPI) {
    * Parse a CAPI search response into a (path -> json) map
    */
   private def parseCapiResponse(responseJson: JsValue): Map[String, JsValue] = {
+    println(responseJson)
```
Comment (LATaylor-guardian) [inline frontend/app/com/gu/recommend/capi/Capi.scala]: Is this meant to be here?

**Comment by LATaylor-guardian** [comment]
:+1: Lets see if it deploys.

File: frontend/app/com/gu/recommend/capi/Capi.scala (hunk 1 above)
Comment (cb372) [inline frontend/app/com/gu/recommend/capi/Capi.scala]: Nope

**Comment by cb372** [comment]
It worked! It's a miracle!

**Comment by LATaylor-guardian** [comment]
:8ball:

**Pull request merged**